### PR TITLE
Add tag filters to SentryLoggingOptions

### PR DESCRIPTION
**Pull request by SeanFeldman**
Allow specifying patterns or regular expressions to filter tags.

### CODE REVIEW

**Comment by SeanFeldman** [comment]
Should `"{OriginalFormat}"` key be guarded to avoid accidental filtering out?

File: src/Sentry.Extensions.Logging/SentryLoggingOptions.cs [hunk 1]
```diff
@@ -48,4 +48,9 @@ public class SentryLoggingOptions : SentryOptions
     /// List of callbacks to be invoked when initializing the SDK
     /// </summary>
     internal Action<Scope>[] ConfigureScopeCallbacks { get; set; } = Array.Empty<Action<Scope>>();
+
+    /// <summary>
+    /// List of substrings or regular expression patterns to filter out tags
+    /// </summary>
+    public SubstringOrRegexPattern[] TagFilters { get; set; } = Array.Empty<SubstringOrRegexPattern>();
```
Comment (mattjohnsonpint) [inline src/Sentry.Extensions.Logging/SentryLoggingOptions.cs:55]:
```suggestion
    public ICollection<SubstringOrRegexPattern> TagFilters { get; set; } = new List<SubstringOrRegexPattern>();
```

Comment (mattjohnsonpint) [inline src/Sentry.Extensions.Logging/SentryLoggingOptions.cs:55]:
```suggestion
    public ICollection<SubstringOrRegexPattern> TagFilters { get; set; } = new List<SubstringOrRegexPattern>();
```

File: src/Sentry.Extensions.Logging/SentryLogger.cs [hunk 2]
```diff
@@ -99,6 +106,12 @@ internal static SentryEvent CreateEvent<TState>(LogLevel logLevel, EventId id, T
                     continue;
                 }
 
+                // TODO: should the code ensure "{OriginalFormat}" is not specified by user and removed?
```
Comment (mattjohnsonpint) [inline src/Sentry.Extensions.Logging/SentryLogger.cs:109]:
```suggestion
```

**Comment by SeanFeldman** [comment]
@mattjohnsonpint, this is ready to be merged.

**Review by mattjohnsonpint — APPROVED** [review]
Looks great.  Thanks! :shipit: :rocket:

**Pull request merged**